xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -324,16 +318,39 @@ bool CProcessorHD::InitProcessor()
         DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709, &supported);
     const bool topLeft = SUCCEEDED(hr) && static_cast<bool>(supported);
 
+    // Check if driver prefers HDR10 LEFT input color space for HDR10 passthrough
+    const DXGI_COLOR_SPACE_TYPE outputCS = DX::Windowing()->UseLimitedColor()
+                                               ? DXGI_COLOR_SPACE_RGB_STUDIO_G2084_NONE_P2020
+                                               : DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020;
+
+    hr = m_pEnumerator1->CheckVideoProcessorFormatConversion(
+        DXGI_FORMAT_P010, DXGI_COLOR_SPACE_YCBCR_STUDIO_G2084_LEFT_P2020, format, outputCS,
+        &supported);
+    const bool HDRleft = SUCCEEDED(hr) && static_cast<bool>(supported);
+
+    hr = m_pEnumerator1->CheckVideoProcessorFormatConversion(
+        DXGI_FORMAT_P010, DXGI_COLOR_SPACE_YCBCR_STUDIO_G2084_TOPLEFT_P2020, format, outputCS,
+        &supported);
+    const bool HDRtopLeft = SUCCEEDED(hr) && static_cast<bool>(supported);
+
```

Comment:
This function is called after switching screen SDR to HDR so assuming output format = back buffer = rgb10 works but doesn't feel very strong.
The log looks odd in SDR for AMD and 8 bit backbuffer ("Use 10 bit for SDR" Auto or off value) because the processor has no support for PQ in B8G8R8A8_UNORM. Odd but not blocking.
Same note as IsHDR10PassthroughSupported(), assuming the input format is always P010.

Reply:
This code is only to determine use Left or TopLeft and if `dxgi_format` is not supported is detected early in `IsHDR10PassthroughSupported()` and this code is not reached anyway.